sympy/vector/functions.py

```diff
@@ -124,22 +124,19 @@ def express(expr, system, system2=None, variables=False):
         return expr
 
 
-def directional_derivative(scalar, vect):
+def directional_derivative(direction_vector, field):
```

Comment:
I like your suggestion. It will be better to turn order of parameters.
There is still question if we should change parameter from `scalar` to `field`? 
`field` argument was earlier in `Vector` class and also with this PR usage of `directional_derivative` is extended to vector field. Leaving `scalar` could be misleading.